scripts/build_proto.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+set -e
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+PROJECT_ROOT="$(cd "${SCRIPT_DIR}/.." && pwd)"
+IMAGE_NAME="omes-ks-gen"
+
+# Always load versions from versions.env
+set -a && source "${PROJECT_ROOT}/versions.env" && set +a
+
+echo "Building Docker image for kitchen-sink-gen..."
+docker build -f "${PROJECT_ROOT}/dockerfiles/proto.Dockerfile" \
+  --build-arg RUST_TOOLCHAIN="${RUST_TOOLCHAIN}" \
+  --build-arg GO_VERSION="${GO_VERSION}" \
+  --build-arg PROTOC_VERSION="${PROTOC_VERSION}" \
+  --build-arg PROTOC_GEN_GO_VERSION="${PROTOC_GEN_GO_VERSION}" \
+  --build-arg NODE_VERSION="${NODE_VERSION}" \
+  -t "${IMAGE_NAME}" "${PROJECT_ROOT}"
+
+echo "Running kitchen-sink-gen build..."
+docker run --rm -v "${PROJECT_ROOT}:/workspace" -w /workspace "${IMAGE_NAME}" bash -c "
```

Comment:
Not a huge fan of mounting the whole project dir into the docker image. This can make for slow loads as it copies everything, and what it writes back out often has weird, incorrect permissions.

That said, this is still easier than before, so I wouldn't block on this, but, if possible might be nice to cleanly copy the result files out.

Reply:
👍 Good idea!

Reply:
Hm, I spent almost an hour on this and it's harder than I thought: I don't want to copy all files into the container (node_modules is large) which requires some filtering, I don't want to specify every single output because that's brittle, but copying all files out of the container also defeats the purpose, doing a diff is harder than it seems, ...

I'll merge for now and we can discuss improvements.